include/deal.II/grid/tria.h

```diff
@@ -1468,6 +1474,18 @@ class Triangulation : public Subscriptor
     distorted_cells;
   };
 
+  /**
+   * If add_periodicity() is called, this variable stores the given
+   * periodic face pairs on level 0 for later access during the
+   * identification of ghost cells for the multigrid hierarchy and for
+   * setting up the periodic_face_map.
+   */
+  std::vector<GridTools::PeriodicFacePair<cell_iterator> > periodic_face_pairs_level_0;
+
+  /**
+   * If add_periodicity() is called, this variable stores the active periodic face pairs.
+   */
+  std::map<std::pair<cell_iterator, unsigned int>, std::pair<std::pair<cell_iterator, unsigned int>, std::bitset<3> > > periodic_face_map;
```

Comment:
@masterleinad It's a bit difficult to read your code because github has removed something, but I think @bangerth referred to choosing `std::map<std::pair<cell_iterator, unsigned int>, std_cxx11::tuple<cell_iterator, unsigned int, std::bitset<3> > >`, so keeping the pair for the key but using tuple for the value because there you need to put three arguments. I would agree with that choice.